lib/common/aspects/resource_prefixer/resource_prefixer.ts

```diff
@@ -0,0 +1,56 @@
+import { CfnResource, IAspect, IConstruct } from "@aws-cdk/core";
+import { DynamoDbCfnTablePrefixer } from "./prefixers/dynamodb_cfn_table_prefixer";
+import { CfnResourcePrefixer } from "./cfn_resource_prefixer";
+import { CfnTable } from "@aws-cdk/aws-dynamodb";
+
+export type Constructor<T> = { new (...args: any[]): T };
+
+export class ResourcePrefixer implements IAspect {
+  private prefix: string;
+  private prefixers: Map<
+    Constructor<CfnResource>,
+    Constructor<CfnResourcePrefixer>[]
+  >;
+
+  constructor(prefix: string) {
+    this.prefix = prefix;
+    this.prefixers = new Map<
+      Constructor<CfnResource>,
+      Constructor<CfnResourcePrefixer>[]
+    >();
+
+    this.registerPrefixer(CfnTable, DynamoDbCfnTablePrefixer);
+  }
+
+  public visit(node: IConstruct): void {
```

Comment:
I'm not sure what this is for?

Reply:
This is the method that needs to be implemented when using the `IAspect` interface, it is called for every node in the CDK stack that is visited as part of the synthesis process